src/batch-get-item.ts

```diff
@@ -0,0 +1,63 @@
+import type { DynamoDB } from "aws-sdk";
+import { KeyAttribute, KeyAttributeToObject } from "./key";
+import { ApplyProjection } from "./projection";
+
+export interface KeysAndAttributes<
+  Item extends object,
+  PartitionKey extends keyof Item,
+  RangeKey extends keyof Item | undefined,
+  Key extends KeyAttribute<Item, PartitionKey, RangeKey>
```

Comment:
This needs to be a tuple so that we preserve ordering of the keys and can narrow each one individually. 

```ts
Keys extends KeyAttribtute<Item, PartitionKey, RangeLey>[]
```

Reply:
We can do this but I'm not sure it really helps much. Dynamo doesn't guarantee ordering of the returned results. Also, the `ProjectionExpression` is applied to all the results so there's no way to project some attributes for key X and some others for key Y.

That said, I am not very familiar with single table design so I'm probably overlooking something